tests/PathMappingRepositoryTest.php

```diff
@@ -131,6 +131,29 @@ public function testResolveFilesystemResource()
         $this->assertEquals(Path::normalize(__DIR__.'/Fixtures/dir5/sub'), $resource->getFilesystemPath());
     }
 
+    public function testMergeMixedParent()
+    {
+        $this->store->set('/', null);
+        $this->store->set('/sub1', array(
+            __DIR__.'/Fixtures/dir6/sub1',
+        ));
+        $this->store->set('/sub1/sub2', null);
+        $this->store->set('/sub1/sub2/sub3/file2', array(
+            __DIR__.'/Fixtures/dir6/sub1/sub2/sub3/file2',
+        ));
+
+        // Get
+        $resource = $this->repo->get('/sub1/sub2/sub3');
+
+        $this->assertInstanceOf('Puli\Repository\Api\Resource\FilesystemResource', $resource);
+        $this->assertEquals(Path::normalize(__DIR__.'/Fixtures/dir6/sub1/sub2/sub3'), $resource->getFilesystemPath());
+
+        // Find
+        $children = $this->repo->listChildren('/sub1/sub2/sub3');
+
+        $this->assertCount(2, $children);
```

Comment:
This test is valid but I think you don't take into account the possibility of multiple parents.

Imagine this store:
``` text
/app => /dir1
/app/blog => /dir2
/app/blog/res => /dir3
```

When you access `/app/blog/res/views/index.html`, the test should check, in this order, all the following paths:

```
/dir3/views/index.html
/dir2/res/views/index.html
/dir1/blog/res/views/index.html
```

And return the first valid one if `$onlyFirst == true` or the list of all of them otherwise.